nmi_mysql/nmi_mysql.py

```diff
@@ -9,9 +9,42 @@
 from queue import Queue
 
 
+CONFIG_KEYS = ['host', 'user', 'password', 'db', 'port']
+MAX_POOL_SIZE = 10
+
+
 class DB(object):
 
-    def __init__(self, conf, autoconnect=False):
+    def __init__(self, conf):
+        for c in CONFIG_KEYS:
+            if c not in conf:
+                raise ValueError('Invalid config object')
+
+        self.conf = conf
+        self.max_pool_size = conf.get('max_pool_size', MAX_POOL_SIZE)
+        self._initialize_pool()
+
+    def _initialize_pool(self):
+        self.pool = Queue(maxsize=self.max_pool_size)
+
+        for _ in range(0, self.max_pool_size):
+            self.pool.put(_Connection(self.conf))
+
+    def query(self, query, params=None):
+        con = self.pool.get()
+        con.connect()
```

Comment:
I think all your analysis is spot on, and I understand why we did it. I also agree with your direction of rewriting the project based on sqlalchemy as it is one of the most used library (which means less bug/more stable/faster bug fixes etc). As for ORM, I agree that we should steer away from it as there's too much unnecessary abstraction (with different db engines for example) and limitations when it comes to querying, let's stay with sqlalchemy Core.